src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -300,76 +273,72 @@ static ucs_status_t uct_rdmacm_cm_create_dummy_qp(struct rdma_cm_id *id,
     return UCS_OK;
 }
 
-static ucs_status_t uct_rdamcm_cm_ep_create_qpn(uct_rdmacm_cm_ep_t *cep,
-                                                struct uct_rdmacm_cm_qp_context *ctx)
+static uint32_t
+uct_rdamcm_cm_ep_create_qpn(uct_rdmacm_cm_ep_t *cep,
+                            struct uct_rdmacm_cm_device_context *ctx)
 {
     ucs_status_t status;
 
     if (ctx->reserved_qpn_support) {
         status = uct_rdamcm_cm_ep_create_reserved_qpn(cep, ctx);      
     } else {
         /* create a dummy qp in order to get a unique qp_num to provide to librdmacm */
-        status = uct_rdmacm_cm_create_dummy_qp(cep->id, ctx->dummy_qp_ctx.cq,
-                                               &cep->qp);
+        status = uct_rdmacm_cm_create_dummy_qp(cep->id, ctx->cq, &cep->qp);
     }
+    ucs_assert_always(status == UCS_OK);
```

Comment:
looks weird, if we fail should return error from this function

Reply:
@yosefe According to a previous comment we changed to return qp number from this function. Seems like it's hard to return both qp number and error status. We can have two choices for this:
1. return some specail value to indicate error, for example 0. This also looks wired.
2. Return status from this function, and add a qp number pointer parameter to store the qp number.
WDYT?

Reply:
i think we should use option (2)